components/browser/toolbar/src/test/java/mozilla/components/browser/toolbar/edit/EditToolbarTest.kt

```diff
@@ -84,6 +84,29 @@ class EditToolbarTest {
         assertFalse(value)
     }
 
+    @Test
+    fun `back press is forwarded to listener`() {
+        var listenerInvoked = false
+        val listener = { listenerInvoked = true }
+
+        val toolbar = BrowserToolbar(context)
+        toolbar.editToolbar.urlView.onAttachedToWindow()
+        toolbar.setOnBackPressListener(listener)
+        toolbar.editToolbar.urlView.setText("https://www.mozilla.org")
+
+        assertEquals(listener, toolbar.editToolbar.onBackPressedListener)
+
+        // TODO this crashes as the shadow library does not have KeyEvent.KEYCODE_BACK
```

Comment:
The Shadow version of KeyEvents does not seem to support KEYCODE_BACK. Is there any way around this?

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/4400286/54060377-59049480-41b1-11e9-9a34-ecddf158e482.png">